### PR TITLE
Stop putting a time caveat on access tokens

### PR DESCRIPTION
**Pull request by richvdh**
The 'time' caveat on the access tokens was something of a lie, since we weren't
enforcing it; more pertinently its presence stops us ever adding useful time
caveats.

Let's move in the right direction by not lying in our caveats.

### CODE REVIEW

File: synapse/api/auth.py [hunk 1]
```diff
@@ -810,6 +810,10 @@ def validate_macaroon(self, macaroon, type_string, verify_expiry, user_id):
         else:
             v.satisfy_general(lambda c: c.startswith("time < "))
```
Comment (NegativeMjark) [inline synapse/api/auth.py]: Might as well remove the ``verify_expiry`` config option while you are at it.

Comment (NegativeMjark) [inline synapse/api/auth.py]: and put a comment explaining why we aren't ever going to check the ``"time < "`` caveats.

Comment (NegativeMjark) [inline synapse/api/auth.py]: But leave the ``v.satisfy_general(lambda c: c.startswith("time < "))`` so that existing tokens will still work.

Comment (NegativeMjark) [inline synapse/api/auth.py]: Except you can't remove the verify_expiry option because it's used in ``validate_short_term_login_token_and_get_user_id``

Comment (NegativeMjark) [inline synapse/api/auth.py]: But you probably want to add a comment to explain what's going on.

**Comment by richvdh** [comment]
test fails seem unrelated.

**Pull request merged**